dvc/tree/git.py

```diff
@@ -114,6 +115,7 @@ def _is_tree_and_contains(obj, path):
                 return True
         return False
 
+    @methodtools.lru_cache(maxsize=1)
     def _git_object_by_path(self, path):
         import git
 
```

Comment:
For the record: might want to check `tree = self.git.tree(self.rev)` logic down below, as there doesn't seem to be a reason to create new tree instance each time we want to check 1 object. Might mitigate this issue.

Reply:
Update: caching tree doesn't replace this optimization (because we still need to walk through it), but actually speeds it up significantly on top. E.g. on master it takes ~4.5sec, with this PR - ~2.5sec and with tree caching ~1.5sec.